core/scale/collection.hpp

```diff
@@ -75,10 +75,13 @@ namespace kagome::scale::collection {
    * same specified type
    * @tparam T collection item type
    * @param stream source stream
+   * @param decode a functor that decodes an object of type T
    * @return decoded collection or error
    */
   template <class T>
-  outcome::result<std::vector<T>> decodeCollection(common::ByteStream &stream) {
+  outcome::result<std::vector<T>> decodeCollection(
+      common::ByteStream &stream,
+      std::function<outcome::result<T>(common::ByteStream &)> decode_f) {
```

Comment:
For what purpose it was done?

Reply:
Because TypeDecoder wasn't accessible in the scope it was required, but scale codec's function decodeExtrinsic was. So I did it in order to be able to use ScaleCodec methods to decode collections of primitives